.editorconfig

```diff
@@ -0,0 +1,18 @@
+[*]
+charset=utf-8
+end_of_line=crlf
```

Comment:
:-1:

This should not be necessary with the git attribute

Reply:
[.editorconfig](https://editorconfig.org/) is an editor/ide agnostic config used for formatting and is mutually exclusive from git attribute. For open source, these are important since a wide range of editors support it even if there is some duplication

Reply:
>We should allow our editors to write the file with their native line ending.

Why? Just curious...

Reply:
That is a platform specific setting and does not make sense on *nix systems where LF is the standard. Git handles the conversion back to what is specified in `.gitattributes`, so unless specific files *must* have a certain line ending (regardless of platform), it doesn't make sense to enforce it at the editor level.

Reply:
This is inconsequential since all modern platforms recognize `\r\n`, especially editors editing js. ¯\_(ツ)_/¯

We can remove it if it unblocks this PR. I think we should have some more consistent guidelines around what constitutes blocking a PR by requesting changes. It seems silly to prevent this PR from going in because of this.

Reply:
Removed for now. Let's discuss it later in meetings.

Reply:
it looks like the editorconfig project has quite a discussion on this. it seems inconclusive but generally the recommendation is the following:

> end_of_line
> Line ending file format (Unix, DOS, Mac)
>
>The values are case insensitive. They will be lowercased by the core library.
>
>NOTE: if you want to use native line endings between different operating systems it is better not to set this option and leave that task to the VCS! In the future we might add a value like native for this scenario (cf #226).

thread: https://github.com/editorconfig/editorconfig/issues/226

Reply:
Sorry.  I did't realize we are using shell scripts in this project.